web/frontend/src/components/slam.vue

```diff
@@ -1,193 +1,221 @@
 
 <script setup lang="ts">
 
-// import { nextTick } from 'vue';
+import { $ref, $computed } from 'vue/macros';
 import { grpc } from '@improbable-eng/grpc-web';
-import { Client, commonApi, slamApi } from '@viamrobotics/sdk';
-import { displayError } from '../lib/error';
+import { Client, commonApi, ServiceError, slamApi } from '@viamrobotics/sdk';
+import { displayError, isServiceError } from '../lib/error';
 import { rcLogConditionally } from '../lib/log';
+import PCD from './pcd/pcd-view.vue';
+import Slam2dRender from './slam-2d-render.vue';
 
-/*
- * import PCD from './pcd/pcd-view.vue';
- * import Slam2dRender from './slam-2d-render.vue';
- */
+type MapAndPose = { map: Uint8Array, pose: commonApi.Pose}
 
 const props = defineProps<{
   name: string
   resources: commonApi.ResourceName.AsObject[]
   client: Client
 }>();
 
-let imageMap = $ref('');
 const selected2dValue = $ref('manual');
+const selected3dValue = $ref('manual');
+let pointCloudUpdateCount = $ref(0);
```

Comment:
Can this be made a boolean to more greatly resemble the switch it is? Instead of +=1, it would just be  `pointCloudUpdateCount = !pointCloudUpdateCount`